src/spikeinterface/core/baserecording.py

```diff
@@ -331,6 +331,11 @@ def get_traces(
         segment_index = self._check_segment_index(segment_index)
         channel_indices = self.ids_to_indices(channel_ids, prefer_slice=True)
         rs = self._recording_segments[segment_index]
+        start_frame = int(start_frame) if start_frame is not None else 0
+        num_samples = rs.get_num_samples()
+        end_frame = int(min(end_frame, num_samples)) if end_frame is not None else num_samples
+        if start_frame < 0:
+            raise ValueError("start_frame cannot be negative")
```

Comment:
```suggestion
```
@h-mayorquin Sam made a good point about making another PR where we add all the checks discussed in #1989 

So let's remove it for now!

Reply:
Yeah, the point was that the other bound will introduce errors I think so no point on getting into that quagmire here. But this one was for free already. 

Anyway, both  work, maybe it is better to do everything in one go as you suggested.